### PR TITLE
Add support for loading selected positions to LazyBlock

### PR DESCRIPTION
**Pull request by sopel39**
When this lands we can add support for selected positions loading to:
1. Dictionary block
2. Page projection
3. Readers (requires extending selected positions to support for monotonic positions)
4. `MergePages`

### CODE REVIEW

File: presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java [hunk 1]
```diff
@@ -275,6 +281,15 @@ public Block getLoadedBlock()
         return lazyData.getFullyLoadedBlock();
     }
 
+    @Override
+    public Block loadSelectedPositions(SelectedPositions selectedPositions)
```
Comment (dain) [inline presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java]: Can we convert get positions and get region to this call, and drop some of the code?

File: presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java [hunk 2]
```diff
@@ -275,6 +281,15 @@ public Block getLoadedBlock()
...
+    {
+        if (isLoaded()) {
+            return getBlock().loadSelectedPositions(selectedPositions);
+        }
+        return lazyData.getFullyLoadedBlock(selectedPositions);
```
Comment (dain) [inline presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java]: I'm curious why this call isn't lazy also.  All of the other calls like this delay computation until actual usage.  If you do decide to make this lazy, it should continue to prune the existing selected positions.

Comment (sopel39) [reply presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java]: I renamed `loadSelectedPositions` to be `getLoadedBlock` overload.

**Comment by sopel39** [comment]
We will use different approach as discussed with @dain

**Pull request closed without merging**